### DISCUSSION TITLE
Using with Google Consent Mode

### DISCUSSION THREAD

**Discussion by axelf** [orestbida/cookieconsent#657] [General]
Hi,
i have created 4 categories. When a category is activated, I transfer the value of each category to the Google Consent Mode snippet to set the values there.
How does the cookie consent script recognize what the user has selected? In this case I don't have any script tags with a data attribute, I just fill in the Google snippet with the values but if the user opens the preferences window again the selected values should be there. How can i do that?

**Comment by axelf** [comment]
Ok, i think i misunderstood something. In the cc_cookie there are all informations about what was selected and what not. I thought it depends on the data attribute in the script tag :-)

**Comment by optimvsjag** [comment]
Hi @axelf, did you set Google Consent Mode in Google Tag Manager? I'm trying and I can't.
We've the cc_cookie value update when the user change your preferencies but I can't update GTM Consent Mode variables.

The cookie value for all accepted is:
%7B%22categories%22%3A%5B%22necessary%22%2C%22functionality%22%2C%22analytics%22%2C%22marketing%22%5D%2C%22revision%22%3A0%2C%22data%22%3Anull%2C%22consentTimestamp%22%3A%222024-03-14T14%3A05%3A28.216Z%22%2C%22consentId%22%3A%225ba1ed95-581d-467b-ad86-a2484cec2ea2%22%2C%22services%22%3A%7B%22necessary%22%3A%5B%5D%2C%22functionality%22%3A%5B%5D%2C%22analytics%22%3A%5B%5D%2C%22marketing%22%3A%5B%5D%7D%2C%22lastConsentTimestamp%22%3A%222024-03-15T12%3A01%3A13.989Z%22%2C%22expirationTime%22%3A1726228873989%7D

I've set my Consent Mode by default all denied and if cc_cookie change should update the consents with this keys:

1. Analitycs: analytics%22%2C%22
2. Marketing: marketing%22%5D%2C%22
3. Functionality: functionality%22%2C%22

I've a variable for every consent like this:
![image](https://github.com/orestbida/cookieconsent/assets/71232507/2a49d494-686b-4528-a6a9-daef683545b1)

And I looking the cookie for changes:
![image](https://github.com/orestbida/cookieconsent/assets/71232507/f15aebbe-322c-4283-89e5-f6b07f375c9f)

But the result is all denied all the time.
![image](https://github.com/orestbida/cookieconsent/assets/71232507/bde2e911-8194-455f-bd05-28d3d63aa55f)

Any idea to solve it? Thanks a lot.

**Reply by axelf** [reply]
Hi @optimvsjag, you have to set up this setting in the domain or account. Go to administration and then container settings. The last option is the consent you have to activate. Then it should work. I used this as a starting point: https://developers.google.com/tag-platform/security/guides/consent?hl=en&consentmode=basic

**Reply by optimvsjag** [reply]
Thanks @axelf  for your answer. I read this guide but I've the same problem... in the GTM guide write:

"Add a function that monitors interactions with your consent buttons and save the interaction using localStorage or cookies. Tip: Make sure to have identifiers for all buttons in your consent banner."

This cookie is the picture showed in my first post, and not runs, GTM not read from cookie status for denied or granted in every consent. And every consent appears 2 times in the same cookie value, so it is confuse and not clear to set in GTM.

When I only accept "Functionality cookies" I've this cc_cookie value: %7B%22categories%22%3A%5B%22necessary%22%2C%22**functionality%22%5D%2C%22**revision%22%3A0%2C%22data%22%3Anull%2C%22consentTimestamp%22%3A%222024-03-18T11%3A09%3A06.861Z%22%2C%22consentId%22%3A%22d2c0f83d-6b90-42de-854d-e4a371b8bd94%22%2C%22services%22%3A%7B%22necessary%22%3A%5B%5D%2C%22**functionality%22%3A%5B%5D%2C%22**analytics%22%3A%5B%5D%2C%22marketing%22%3A%5B%5D%7D%2C%22lastConsentTimestamp%22%3A%222024-03-18T11%3A09%3A06.861Z%22%2C%22expirationTime%22%3A1726484946861%7D

For Functionality cookie I am searching for "functionality%22%2C%22", in the cc_cookie value we've 2 functionality values but anyone is this so the cc_cookie have different values for only one consent accepted or all consents accepted. So it's difficult to set GTM conditions.

**Reply by optimvsjag** [reply]
I was able to solve it with this topic: https://github.com/orestbida/cookieconsent/issues/110 and this tutorial https://uninterrupted.tech/blog/manage-user-cookie-consent-with-google-tag-manager-a-step-by-step-guide/ adding some JavaScript to read the cc_cookie content.
